src/styled.js

```diff
@@ -43,9 +43,7 @@ function styled(tag, forwardRef) {
                 css.apply(_ctx, _args) + (_previousClassName ? ' ' + _previousClassName : '');
 
             // If the forwardRef fun is defined we have the ref
-            if (forwardRef) {
-                _props.ref = ref;
-            }
+            _props.ref = forwardRef && ref;
```

Comment:
Unfortunately, this will define `.ref` as a boolean and `React` expects `.ref` to be undefined when the component is not proxied through `forwardRef`. This works fine in Preact but not in React.